lib/util.rb

```diff
@@ -211,4 +221,51 @@ def self.project_data(project)
     end
     project_data
   end
+
+  # this method takes a hash of data and two dates. It takes the data that falls
+  # between the two dates, then returns the sum of their measurements
+  def self.data_in_date_range(input_hash, start_date, end_date)
```

Comment:
I think we're doing unnecessary work here - when we fetch the data from the plugin's report method , we can send a date range, so we only ever fetch the data in that range rather than fetching it all and pulling the relevant range in a new method. When the data is fetched for each project's resource's measurement plugin (project_data method in Report class, line 209 util.rb), we should call the report method with the given date range.

To do that, add startdate, enddate parameters to the project_data method (default nil), and call project_data from the route with the dates you receive from the form.

Then this method only needs to sum up the given data, since the data should only contain the correct time range.